lib/pure/includes/osenv.nim

```diff
@@ -201,27 +102,17 @@ else:
     ## * `existsEnv proc <#existsEnv,string>`_
     ## * `delEnv proc <#delEnv,string>`_
     ## * `envPairs iterator <#envPairs.i>`_
-
-    # Note: by storing the string in the environment sequence,
-    # we guarantee that we don't free the memory before the program
-    # ends (this is needed for POSIX compliance). It is also needed so that
-    # the process itself may access its modified environment variables!
     when nimvm:
       discard "built into the compiler"
     else:
-      var indx = findEnvVar(key)
-      if indx >= 0:
-        environment[indx] = key & '=' & val
-      else:
-        add environment, (key & '=' & val)
-        indx = high(environment)
       when defined(windows) and not defined(nimscript):
```

Comment:
remove `and not defined(nimscript)` which is useless (`putEnv` is a `vmops` since at least the version of nim that is used to bootstrap nim so this is a noop)

(see vmops.nim)

likewise in other similar instances.
Feel free to ask questions if anything is unclear

Reply:
So `if defined nimvm then defined nimscript`, effectively?

Reply:
`if defined nimvm` cannot work (`when nimvm` is special in the compiler). In fact you don't even need `when nimvm:` I think, because vmops will intercept this call so the body of this proc won't execute in VM (whether in regular VM or in nimscript).

in summary:
if a proc is a vmops (see vmops.nim), the body won't be evaluated at CT (including in nimscript)
there's a complication for boostrap, if the vmops was added recently (more recently than the version that bootstraps nim) AND if the code is used at CT by koch.nim or other code needed during bootstrap, but IIUC that's not the case here. Let me know if still unclear.

i think all you need here is:
```nim
  proc putEnv*(key, val: string) {.tags: [WriteEnvEffect].} =
    ## comments...
    when defined(windows):
    else: ...
```

Reply:
Great, I'll change it and see if CI still succeeds.